src/main/java/com/klodnicki/Bike/v2/service/RentBikeService.java

```diff
@@ -0,0 +1,174 @@
+package com.klodnicki.Bike.v2.service;
+
+
+import com.klodnicki.Bike.v2.DTO.bike.BikeForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentRequestDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentResponseDTO;
+import com.klodnicki.Bike.v2.DTO.user.UserForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.model.RentRequest;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.model.entity.ChargingStation;
+import com.klodnicki.Bike.v2.model.entity.Rent;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.RentRepository;
+import com.klodnicki.Bike.v2.service.interfacee.*;
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class RentBikeService implements GenericRentBikeService {
+
+    private final GenericBikeService bikeService;
+    private final GenericChargingStationService chargingStationService;
+    private final GenericUserService userService;
+    private final ModelMapper modelMapper;
+    private final RentRepository rentRepository;
+
+    @Override
+    public RentResponseDTO updateRent(Long id, RentRequestDTO rentRequestDTO) {
+        Rent rent = findRentById(id);
+
+        if (rentRequestDTO.getDaysOfRent() > 0) {
+            rent.setDaysOfRent(rentRequestDTO.getDaysOfRent());
+        }
```

Comment:
Could you please explain why this piece of code is required here?

Reply:
yes, days of rent is a primitive data type (int), so if it will be skipped in a JSON request  then it will be set to 0.

Reply:
So this is a perfect place to not use primitive type.
Or maybe I'm confused of what should "daysOfRent" means - what is the purpose of this field in the Rent entity?